mempool/src/core_mempool/transaction.rs

```diff
@@ -15,6 +18,17 @@ use std::{
 /// Estimated per-txn size minus the raw transaction
 pub const TXN_FIXED_ESTIMATED_BYTES: usize = size_of::<MempoolTransaction>();
 
+// This is the sequence number for an account.
+// For the sender of regular transactions, the sequence number is required.
+// For the sender of orderless transactions, we don't calculate the sequence number.
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash, Serialize, Deserialize)]
+pub enum AccountSequenceNumberInfo {
+    // Question[Orderless]: Please suggest some better names.
+    // Is there a better way to represent this?
```

Comment:
why not just use replay protector here instead of a new struct?

Reply:
When a seq number based transaction is to be committed, I am using `AccountSequenceNumberInfo ::seq_num_split_off` function to figure out the list of transactions that need to be purged from mempool. This data structure makes it convenient. Please let me know if you have other thoughts.